diff --git a/node_modules/minecraft-protocol/src/client/chat.js b/node_modules/minecraft-protocol/src/client/chat.js
index 8d0869b..cc7433a 100644
--- a/node_modules/minecraft-protocol/src/client/chat.js
+++ b/node_modules/minecraft-protocol/src/client/chat.js
@@ -3,7 +3,7 @@ const concat = require('../transforms/binaryStream').concat
 const { processNbtMessage } = require('prismarine-chat')
 const messageExpireTime = 420000 // 7 minutes (ms)
 
-function isFormatted (message) {
+function isFormatted(message) {
   // This should match the ChatComponent.isDecorated function from Vanilla
   try {
     const comp = JSON.parse(message)
@@ -37,7 +37,7 @@ module.exports = function (client, options) {
   // This stores last 1024 inbound messages for report lookup
   client._lastChatHistory = new class extends Array {
     capacity = 1024
-    push (e) {
+    push(e) {
       super.push(e)
       if (this.length > this.capacity) {
         this.shift()
@@ -45,7 +45,7 @@ module.exports = function (client, options) {
     }
   }()
 
-  function updateAndValidateSession (uuid, message, currentSignature, index, previousMessages, salt, timestamp) {
+  function updateAndValidateSession(uuid, message, currentSignature, index, previousMessages, salt, timestamp) {
     const player = client._players[uuid]
 
     if (player && player.hasChainIntegrity) {
@@ -58,8 +58,13 @@ module.exports = function (client, options) {
 
       if (player.hasChainIntegrity) {
         const length = Buffer.byteLength(message, 'utf8')
-        const acknowledgements = previousMessages.length > 0 ? ['i32', previousMessages.length, 'buffer', Buffer.concat(...previousMessages.map(msg => msg.signature || client._signatureCache[msg.id]))] : ['i32', 0]
+        const ackBufs = previousMessages
+          .map(msg => msg.signature || client._signatureCache[msg.id])
+          .filter(Boolean); // Remove undefined entries
 
+        const acknowledgements = ackBufs.length > 0
+          ? ['i32', ackBufs.length, 'buffer', Buffer.concat(ackBufs)]
+          : ['i32', 0];
         const signable = concat('i32', 1, 'UUID', uuid, 'UUID', player.sessionUuid, 'i32', index, 'i64', salt, 'i64', timestamp / 1000n, 'i32', length, 'pstring', message, ...acknowledgements)
 
         player.hasChainIntegrity = crypto.verify('RSA-SHA256', signable, player.publicKey, currentSignature)
@@ -71,7 +76,7 @@ module.exports = function (client, options) {
     return false
   }
 
-  function updateAndValidateChat (uuid, previousSignature, currentSignature, payload) {
+  function updateAndValidateChat(uuid, previousSignature, currentSignature, payload) {
     // Get the player information
     const player = client._players[uuid]
     if (player && player.hasChainIntegrity) {
@@ -298,7 +303,7 @@ module.exports = function (client, options) {
     for (const commandNode of nodes[0].children) {
       const node = nodes[commandNode]
       const commandName = node.extraNodeData.name
-      function visit (node, depth = 0) {
+      function visit(node, depth = 0) {
         const name = node.extraNodeData.name
         if (node.extraNodeData.parser === 'minecraft:message') {
           sliceIndexForMessage[commandName] = [name, depth]
@@ -311,7 +316,7 @@ module.exports = function (client, options) {
     }
   })
 
-  function signaturesForCommand (string, ts, salt, preview, acknowledgements) {
+  function signaturesForCommand(string, ts, salt, preview, acknowledgements) {
     const signatures = []
     const slices = string.split(' ')
     if (sliceIndexForMessage[slices[0]]) {
@@ -329,7 +334,7 @@ module.exports = function (client, options) {
   let pendingChatRequest
   let lastPreviewRequestId = 0
 
-  function getAcknowledgements () {
+  function getAcknowledgements() {
     let acc = 0
     const acknowledgements = []
 
@@ -495,7 +500,7 @@ module.exports = function (client, options) {
   client.reportPlayer = (uuid, reason, reportedSignatures, comments) => {
     const evidence = []
 
-    function addEvidence (entry) {
+    function addEvidence(entry) {
       evidence.push({
         previousHeaderSignature: entry.previousSignature,
         uuid: entry.senderUuid,
@@ -532,7 +537,7 @@ class SignatureCache extends Array {
   capacity = 128
   index = 0
 
-  push (e) {
+  push(e) {
     if (!e) return
 
     this[this.index++] = e
@@ -545,7 +550,7 @@ class LastSeenMessagesWithInvalidation extends Array {
   offset = 0
   pending = 0
 
-  push (e) {
+  push(e) {
     if (!e) return false
 
     this[this.offset] = { pending: true, signature: e }
@@ -559,7 +564,7 @@ class LastSeenMessages extends Array {
   capacity = 5
   pending = 0
 
-  push (e) {
+  push(e) {
     if (!e || !e.signature || e.signature.length === 0) return false // We do not acknowledge unsigned messages
 
     // Insert a new entry at the top and shift everything to the right
